src/Ocelot/Authorization/ScopesAuthorizer.cs

```diff
@@ -28,7 +31,14 @@ public Response<bool> Authorize(ClaimsPrincipal claimsPrincipal, List<string> ro
             return new ErrorResponse<bool>(values.Errors);
         }
 
-        var userScopes = values.Data;
+        IList<string> userScopes = values.Data;
+
+        // There should not be more than one scope claim that has space-separated value by design
+        // Some providers use array value some space-separated value but not both
+        if (userScopes.Count == 1 && userScopes[0].Contains(' '))

```

Comment:
```suggestion
        if (userScopes.Count == 1 && userScopes[0].Contains(SpaceChar))
```

Reply:
My case was [auth0](https://auth0.com). The [RFC ](https://www.rfc-editor.org/rfc/rfc8693#section-4.2) states the scope claim is a space-separated value but sadly implementations varies as I can see.